### PR TITLE
Add support for all multi-byte currency symbols

### PR DESCRIPTION
**Pull request by sumitj824**
### Description

Earlier Babel only supported dollar currency symbol because other currency symbols are multi-byte characters because in the parser we had support only for single byte currency symbols. With this commit we have modified the parser in order to support other currency symbols.
 
Task: BABEL-2636
authored-by: Sumit Jaiswal <sumiji@amazon.com>
Signed-off-by: Kushaal Shroff <kushaal@amazon.com>

[List any issues this PR will resolve]

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/Makefile [hunk 1]
```diff
@@ -94,15 +94,15 @@ UPGRADES = $(patsubst sql/upgrades/%.sql,sql/%.sql,$(wildcard sql/upgrades/*.sql
 REGRESS = test/babel_like
 REGRESS += test/babel_219
 REGRESS += test/babel_init
-REGRESS += test/babel_select_distinct_top
+# REGRESS += test/babel_select_distinct_top
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/Makefile]: why is this commented?

File: contrib/babelfishpg_tsql/expected/test/babel_datatype.out [hunk 2]
```diff
@@ -727,6 +792,9 @@ WARNING:  TIMESTAMP(7) WITH TIME ZONE precision reduced to maximum allowed, 6
 LINE 1: create table testing1(ts DATETIME, tstz DATETIMEOFFSET(7));
                                                 ^
 WARNING:  TIMESTAMP(7) WITH TIME ZONE precision reduced to maximum allowed, 6
+LINE 1: create table testing1(ts DATETIME, tstz DATETIMEOFFSET(7));
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/expected/test/babel_datatype.out:795]: why this diff?

File: contrib/babelfishpg_tsql/expected/test/babel_datatype.out [hunk 3]
```diff
@@ -932,13 +1003,21 @@ select CAST('£' AS sys.varchar(1));		-- allowed
 (1 row)
 
 select CAST('😀' AS char(1));			-- not allowed
-ERROR:  value too long for type character(1) as UTF16 output
+ bpchar
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/expected/test/babel_datatype.out]: why this diff?

File: contrib/babelfishpg_tsql/src/backend_parser/scan-tsql-rule.l [hunk 4]
```diff
@@ -101,12 +101,24 @@
 					return TSQL_ATAT;
 				}
 
-<tsql>{tsql_money}	{
+<tsql>{tsql_money_with_one_byte} {
 					SET_YYLLOC();
-					yylval->str = pstrdup(yytext+1); /* stripping currency symbol */
+					yylval->str = pstrdup(yytext+1); /* stripping currency symbol with one byte*/
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/src/backend_parser/scan-tsql-rule.l]: space before */

File: contrib/babelfishpg_tsql/Makefile (hunk 1 above)
Comment (sumitj824) [reply contrib/babelfishpg_tsql/Makefile]: need to revert it

Comment (sumitj824) [reply contrib/babelfishpg_tsql/Makefile:97]: It exist in internal so I think I need not comment it. I will revert it

File: contrib/babelfishpg_tsql/expected/test/babel_datatype.out (hunk 3 above)
Comment (sumitj824) [reply contrib/babelfishpg_tsql/expected/test/babel_datatype.out]: Earlier it was not supported so the old expected file expect error

File: contrib/babelfishpg_tsql/expected/test/babel_datatype.out (hunk 2 above)
Comment (sumitj824) [reply contrib/babelfishpg_tsql/expected/test/babel_datatype.out:795]: I have talked to Rishabh and Jason this diff is expected

**Review by surendravishnoi — DISMISSED** [review]

**Review by KushaalShroff — APPROVED** [review]

**Review by surendravishnoi — APPROVED** [review]

**Pull request merged**